### PR TITLE
Update Ektorp library to 1.4.2

### PR DESCRIPTION
**Pull request by RicardoBelchior**
Updated ektorp library version to 1.4.2 and implemented missing methods
from interface HttpClient and HttpResponse.

### CODE REVIEW

**Comment by RicardoBelchior** [comment]
not sure about the new post and put methods, since i'm not familiared with the http intrincacies

File: build.gradle [hunk 1]
```diff
@@ -62,28 +62,28 @@ task uploadArchivesWrapper(dependsOn: createMavenDirectory) << {
...
+//                    license {
+//                        name 'Couchbase Community Edition License Agreement'
+//                        url 'http://www.couchbase.com/agreement/community'
+//                        distribution 'repo'
+//                    }
+//                }
+//            }
+//        }
+//    }
+//}
```
Comment (tleyden) [inline build.gradle]: any reason to comment the uploadArchives directive?  does this interfere with anything?

Comment (hideki) [inline build.gradle]: These lines of codes cause build error.

**Comment by hideki** [comment]
Hi @RicardoBelchior,

I fixed some of your PR and created new branch because I don't have permission to push to the branch under your account. Could you please review? branch: https://github.com/couchbaselabs/couchbase-lite-android-ektorp/tree/feature/issue_upgrade_ektorp_version

Thanks!

File: build.gradle (hunk 1 above)
Comment (RicardoBelchior) [inline build.gradle]: yes, that was it

**Pull request merged**